cpp/test/lap/lap.cu

```diff
@@ -0,0 +1,167 @@
+ *
+ */
+#include <gtest/gtest.h>
+
+#include <omp.h>
+#include <iostream>
+#include <raft/lap/lap.cuh>
+#include <random>
+
+#define PROBLEMSIZE 1000  // Number of rows/columns
+#define BATCHSIZE 10      // Number of problems in the batch
+#define COSTRANGE 1000
+#define PROBLEMCOUNT 1
+#define REPETITIONS 1
+
+#define SEED 01010001
+
+std::default_random_engine generator(SEED);
+
+namespace raft {
+
+// Function for generating problem with uniformly distributed integer costs between [0, COSTRANGE].
+template <typename weight_t>
+void generateProblem(weight_t *cost_matrix, int SP, int N, int costrange) {
+  long N2 = SP * N * N;
+
+  std::uniform_int_distribution<int> distribution(0, costrange);
+
+  for (long i = 0; i < N2; i++) {
+    int val = distribution(generator);
+    cost_matrix[i] = (weight_t)val;
+  }
+}
+
+template <typename vertex_t, typename weight_t>
+void hungarian_test(int problemsize, int costrange, int problemcount,
+                    int repetitions, int batchsize, bool verbose = false) {
+  raft::handle_t handle;
+
+  weight_t *h_cost = new weight_t[batchsize * problemsize * problemsize];
+
+  for (int j = 0; j < problemcount; j++) {
+    generateProblem(h_cost, batchsize, problemsize, costrange);
+
+    raft::mr::device::buffer<weight_t> elements_v(
+      handle.get_device_allocator(), handle.get_stream(),
+      batchsize * problemsize * problemsize);
+    raft::mr::device::buffer<vertex_t> row_assignment_v(
+      handle.get_device_allocator(), handle.get_stream(),
+      batchsize * problemsize);
+    raft::mr::device::buffer<vertex_t> col_assignment_v(
+      handle.get_device_allocator(), handle.get_stream(),
+      batchsize * problemsize);
+
+    raft::update_device(elements_v.data(), h_cost,
+                        batchsize * problemsize * problemsize,
+                        handle.get_stream());
+
+    for (int i = 0; i < repetitions; i++) {
+      float start = omp_get_wtime();
+
+      // Create an instance of LinearAssignmentProblem using problem size, number of subproblems
+      raft::lap::LinearAssignmentProblem<vertex_t, weight_t> lpx(
+        handle, problemsize, batchsize);
+
+      // Solve LAP(s) for given cost matrix
+      lpx.solve(elements_v.data(), row_assignment_v.data(),
+                col_assignment_v.data());
+
+      float end = omp_get_wtime();
+
+      float total_time = (end - start);
+
+      if (verbose) {
+        // Use getPrimalObjectiveValue and getDualObjectiveValue APIs to get primal and dual objectives. At optimality both values should match.
+        for (int k = 0; k < batchsize; k++) {
+          std::cout << j << ":" << i << ":" << k << ":"
+                    << lpx.getPrimalObjectiveValue(k) << ":"
+                    << lpx.getDualObjectiveValue(k) << ":" << total_time
+                    << std::endl;
+        }
+      }
+
+      //			Use getAssignmentVector API to get the optimal row assignments for specified problem id.
+      //			Example is shown below.
```

Comment:
small question, why is the example here?

Reply:
This was code from a third party and I had accidentally left these existing comments in the code.  I have deleted these comments.